src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Areas/AreaGenerator.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.VisualStudio.Web.CodeGeneration;
+using Microsoft.VisualStudio.Web.CodeGeneration.CommandLine;
+using Microsoft.VisualStudio.Web.CodeGeneration.DotNet;
+using Microsoft.VisualStudio.Web.CodeGenerators.Mvc.Dependency;
+
+namespace Microsoft.VisualStudio.Web.CodeGenerators.Mvc.Areas
+{
+    [Alias("area")]
+    public class AreaGenerator : ICodeGenerator
+    {
+        private static readonly string[] AreaFolders = new string[]
+        {
+            "Controllers",
+            "Models",
+            "Data",
+            "Views"
+        };
+
+        private IServiceProvider _serviceProvider { get; set; }
+        private IApplicationInfo _appInfo { get; set; }
+        private ILogger _logger { get; set; }
+        private IModelTypesLocator _modelTypesLocator { get; set; }
+
+        public AreaGenerator(IApplicationInfo applicationInfo,
+            IServiceProvider serviceProvider,
+            IModelTypesLocator modelTypesLocator,
+            ILogger logger)
+        {
+            if(serviceProvider == null)
```

Comment:
>if(s [](start = 12, length = 4)

Format document, some times you have spaces in between keywords and ( , some times not. Use default formatting settings (Edit\Advanced\Format document.. ?)